geometric_kernels/spaces/hyperbolic.py

```diff
@@ -0,0 +1,82 @@
+"""
+Hyperbolic space.
+"""
+
+from typing import Optional
+
+import geomstats as gs
+import lab as B
+
+from geometric_kernels.lab_extras import cosh, trapz
+from geometric_kernels.spaces import Space
+
+
+class Hyperbolic(Space, gs.geometry.hyperboloid.Hyperboloid):
+    r"""
+    Hyperbolic manifold.
+    """
+
+    def __init__(self, dim=1):
+        super().__init__(dim=dim)
+
+    @property
+    def dimension(self) -> int:
+        return self.dim
+
+    def distance(
+        self, x1: B.Numeric, x2: B.Numeric, diag: Optional[bool] = False
+    ) -> B.Numeric:
+        assert B.all(self.belongs(x1)) and B.all(self.belongs(x2))
+
+        return self.metric.dist(x1, x2)
+
+    def heat_kernel(
+        self, distance: B.Numeric, t: B.Numeric, num_points: int = 100
+    ) -> B.Numeric:
+        """
+        Compute the heat kernel associated with the space.
```

Comment:
Should we precise that we use Millson's formula here and refer to the corresponding paper?

[1] A. Grigoryan and M. Noguchi. The heat kernel on hyperbolic space.Bulletin of the London Mathematical Society, 30(6):643–650, 1998.